### PR TITLE
chore: release assets for multiple platforms

### PR DESCRIPTION
**Pull request by suzuki-shunsuke**
https://github.com/slsa-framework/slsa-verifier/pull/421#discussion_r1063056985

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -32,23 +32,31 @@ jobs:
...
     uses: slsa-framework/slsa-github-generator/.github/workflows/builder_go_slsa3.yml@v1.4.0
     with:
       go-version: 1.18
-      config-file: .github/config-release.yml
+      config-file: .slsa-goreleaser-${{matrix.os}}-${{matrix.arch}}.yml
       compile-builder: true
       evaluated-envs: "VERSION:${{needs.args.outputs.version}}"
 
   # In case this fails, e.g. build configuration changes, file an issue in slsa-verifier
   if-failed:
```
Comment (ianlewis) [inline .github/workflows/release.yml:56]: @laurentsimon @asraa Do we even need this in the release workflow? It's not a pre-submit so I don't think we need it to create GitHub issues?

**Review by ianlewis — APPROVED** [review]
LGTM

**Comment by suzuki-shunsuke** [comment]
I tested, then SLSA Provenance was created successfully.

- https://github.com/suzuki-shunsuke/slsa-verifier/releases/tag/v2.0.2-rc.2

The files were created per platform separately.

* slsa-verifier-linux-amd64.intoto.jsonl
* slsa-verifier-linux-arm64.intoto.jsonl

I think ideally Provenance files should be merged into one file `multiple.intoto.jsonl`.

**Comment by ianlewis** [comment]
> I think ideally Provenance files should be merged into one file `multiple.intoto.jsonl`.

Yeah, ideally we'd do that but I think it's ok for now. See https://github.com/slsa-framework/slsa-github-generator/issues/63

**Comment by suzuki-shunsuke** [comment]
I added macOS and windows.

**Comment by suzuki-shunsuke** [comment]
I tested and it worked well.
https://github.com/suzuki-shunsuke/slsa-verifier/actions/runs/3855600047
https://github.com/suzuki-shunsuke/slsa-verifier/releases/tag/v2.0.2-rc.3

File: .github/workflows/release.yml (hunk 1 above)
Comment (laurentsimon) [reply .github/workflows/release.yml:58]: I think you're right we can remove the `if-failed`. @suzuki-shunsuke please remove it and let's merge.

**Review by laurentsimon — APPROVED** [review]
Thank you! Looks great!

File: .github/workflows/release.yml (hunk 1 above)
Comment (suzuki-shunsuke) [reply .github/workflows/release.yml]: Removed. bec4a0b8f053af283c72deffffcb5afb3637193f

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -32,28 +32,24 @@ jobs:
...
+          - arm64
     permissions:
       actions: read # For the detection of GitHub Actions environment.
       id-token: write # For signing.
       contents: write # For asset uploads.
     uses: slsa-framework/slsa-github-generator/.github/workflows/builder_go_slsa3.yml@v1.4.0
     with:
       go-version: 1.18
-      config-file: .github/config-release.yml
+      config-file: .slsa-goreleaser-${{matrix.os}}-${{matrix.arch}}.yml
```
Comment (laurentsimon) [inline .github/workflows/release.yml:53]: Could you move the config files into a new .goreleaser folder? Sorry, I did not catch this earlier. Thanks!

Comment (suzuki-shunsuke) [reply .github/workflows/release.yml:53]: These configuration files are not for GoReleaser, so I prefer `.slsa-goreleaser` than `.goreleaser`.
How about `.slsa-goreleaser/${{matrix.os}}-${{matrix.arch}}.yml`?
What do you think?

Comment (laurentsimon) [reply .github/workflows/release.yml:53]: SGTM

Comment (suzuki-shunsuke) [reply .github/workflows/release.yml]: Moved. 477c130cfa318e7c5c01d9f75cb8810d2a2d6b41

Comment (suzuki-shunsuke) [reply .github/workflows/release.yml]: It works well.
https://github.com/suzuki-shunsuke/slsa-verifier/releases/tag/v2.0.2-rc.4
https://github.com/suzuki-shunsuke/slsa-verifier/actions/runs/3859552606

**Pull request merged**